ncbi/src/main/java/org/reactome/release/NCBIGene.java

```diff
@@ -0,0 +1,184 @@
+		for (NCBIEntry ncbiEntry : ncbiEntries) {
+			ncbiGeneLogger.info("Working on " + ncbiEntry.getUniprotAccession());
+
+			Set<ReactomeEvent> topLevelPathways = ncbiEntry.getTopLevelPathways(graphDBSession);
+			if (topLevelPathways.isEmpty()) {
+				String errorMessage = ncbiEntry.getUniprotDisplayName() +
+									  " participates in Event(s) but no top Pathway can be found, i.e. there seem to be a pathway" +
+									  " which contains or is an instance of itself.\n";
+
+				Files.write(geneErrorFilePath, errorMessage.getBytes(), StandardOpenOption.APPEND);
+				continue;
+			}
+
+			for (String ncbiGeneId : ncbiEntry.getNcbiGeneIds()) {
+				ncbiGeneXMLNodeStrings.add(ncbiEntry.getEntityLinkXML(ncbiGeneId));
+
+				for (ReactomeEvent topLevelPathway : topLevelPathways) {
+					ncbiGeneXMLNodeStrings.add(ncbiEntry.getEventLinkXML(ncbiGeneId, topLevelPathway));
+				}
+			}
+
+			ncbiGeneLogger.info("Finished with " + ncbiEntry.getUniprotAccession());
+		}
+
+		int fileCount = 0;
+		for (Set<String> ncbiGeneXMLNodeStringsSubSet : Utilities.splitSet(ncbiGeneXMLNodeStrings, numGeneXMLFiles)) {
+			Path geneXMLFilePath = getGeneXMLFilePath(++fileCount);
+			Files.deleteIfExists(geneXMLFilePath);
+			Files.createFile(geneXMLFilePath);
+
+			logger.info("Generating " + geneXMLFilePath.getFileName());
+
+			appendWithNewLine(getXMLHeader(), geneXMLFilePath);
+			appendWithNewLine(getOpenRootTag(), geneXMLFilePath);
+			for (String ncbiGeneXMLNodeString : ncbiGeneXMLNodeStringsSubSet) {
+				appendWithNewLine(ncbiGeneXMLNodeString, geneXMLFilePath);
+			}
+
+			appendWithNewLine(getCloseRootTag(), geneXMLFilePath);
+		}
+
+		logger.info("Finished writing gene XML file(s)");
+	}
+
+	private Path getProteinFilePath() {
+		return Paths.get(outputDir, "proteins_version" + reactomeVersion);
+	}
+
+	private Path getGeneXMLFilePath(int fileCount) {
+		String fileName = "gene_reactome" + reactomeVersion + "-" + fileCount + ".xml";
+		return Paths.get(outputDir, fileName);
+	}
+
+	private Path getGeneErrorFilePath() {
+		return Paths.get(outputDir, "geneentrez_" + reactomeVersion + ".err");
+	}
+
+	/**
+	 * Returns the XML header for the NCBI Gene file containing the Reactome entity and event base URLs
+	 * @return XML header as String
+	 */
+	public static String getXMLHeader() {
+		return String.join(System.lineSeparator(),
+						   "<?xml version=\"1.0\"?>",
+						   "<!DOCTYPE LinkSet PUBLIC \"-//NLM//DTD LinkOut 1.0//EN\"",
+						   "\"http://www.ncbi.nlm.nih.gov/entrez/linkout/doc/LinkOut.dtd\"",
+						   "[",
+						   "<!ENTITY entity.base.url \"" + ReactomeConstants.UNIPROT_QUERY_URL + "\">",
+						   "<!ENTITY event.base.url \"" + ReactomeConstants.PATHWAY_BROWSER_URL + "\">",
+						   "]>"
+		).concat(System.lineSeparator());
+	}
+
+	/**
+	 * Returns the opening XML root level tag for the NCBI Gene file
+	 * @return XML opening root level tag as String
+	 */
+	public static String getOpenRootTag() {
```

Comment:
There are two different classes that have a getOpenRootTag method. This suggests there is a common interface lurking in the design...

Reply:
Yup, this has been removed in the generation of XML for one of the export files.  I have to re-work the generation of XML here as well.  I will commit those changes soon before requesting a second round of review.